fpnge.cc

```diff
@@ -754,24 +751,22 @@ static FORCE_INLINE void WriteBits(MIVEC nbits, MIVEC bits_lo, MIVEC bits_hi,
   bits1_32_hi = MM(cvtps_epi32)(_mm_castsi128_ps(bits1_32_hi));
 #endif
 
-  auto nbits0_32 = MM(add_epi32)(nbits0_32_lo, nbits0_32_hi);
-  auto nbits1_32 = MM(add_epi32)(nbits1_32_lo, nbits1_32_hi);
+  nbits0 = MM(madd_epi16)(nbits0, MM(set1_epi16)(1));
```

Comment:
I guess you tried it, but wouldn't hadd_epi16 be faster here?

Reply:
Main issue with hadd is that it shifts the positions of everything, whilst madd_epi16 doesn't.  
The latter is also generally faster, as most CPUs implement hadd as multiple uOps, whilst madd is usually 1 uOp.